src/layout/HeaderSearch.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { ChangeEvent, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useDebounceFn } from 'ahooks';
+import { TextInput } from 'carbon-components-react';
+
+import './HeaderSearch.scss';
+
+function HeaderSearch({ }: { }) {
+  const location = useLocation();
+  const { pathname } = location;
+  const pathKey = '/search/'
+  const initKeyword = pathname.includes(pathKey) ? pathname.substr(pathname.indexOf(pathKey) + pathKey.length) : '';
+
+  const [keyword, setKeyword] = useState(initKeyword);
+  const history = useHistory();
+
+  const handleSearchThrottled = useDebounceFn(
+    () => {
+      if (!keyword || keyword === '') {
+        history.push('/');
+      } else {
+        history.push(`${pathKey}${keyword}`);
```

Comment:
maybe use `replace` instead, otherwise you cannot click "back" any more due to too many histories..